### ISSUE TITLE
Base for gulp.src is empty if working directory has braces

### ISSUE THREAD

**Issue by sydorivskyi** [gulpjs/gulp#1744]
Relative gulp version: 4

Create any directory with brackets i.e "**gulp (brackets)**"
- create readme.md file
- create simple gulpfile

```
const gulp = require('gulp');

gulp.task('default', function () {
  return gulp.src('readme.md')
    .pipe(gulp.dest('dist'));
})
```
- run gulp

**Actual result:**
In directory **dest** we can see **readme.md** inside "**gulp (brackets)**" directory

**Expected result:**
In directory **dest** we can see **readme.md** file

**Comment by heikki** [comment]
Minimal repo https://github.com/heikki/gulp-1744

<img width="639" alt="screen shot 2016-08-15 at 01 50 09" src="https://cloud.githubusercontent.com/assets/82735/17652652/e6ee997a-628a-11e6-8ecd-683c99c15ce7.png">

**Comment by phated** [comment]
Are parentheses generally valid in filenames?  Does this fall into a glob issue?  I wonder if they use parens as a special character.

**Comment by stevelacy** [comment]
Tested latest `glob` and `glob-stream`, both return the file path correctly.

https://github.com/stevelacy/glob-parens-test

![selection_0472016dt](https://cloud.githubusercontent.com/assets/5648999/17782077/c78e19c2-6526-11e6-82e4-ada6a8a85e23.png)

**Comment by phated** [comment]
@stevelacy does it work with a space between `gulp` and the first paren?

**Comment by stevelacy** [comment]
The output from glob-stream:

``
{ 
  cwd: '/home/sl/Projects/node/glob-parens-test (parens)',
  base: '/home/sl/Projects/node/',
  path: '/home/sl/Projects/node/glob-parens-test (parens)/folder/gulp (brackets)/readme.md'}

```

**Comment by phated** [comment]
Thanks @stevelacy - I'm really confused about this then.  @heikki did you confirm that it's a problem or just create the repo?

**Comment by heikki** [comment]
@phated repo task creates `gulp (brackets)` dir like in the issue description, I didn't debug it further

**Comment by lukaskostial** [comment]
I'm having the same issue on Windows as our Jenkins in located in C:\Program Files (x86)\jenkins and it seems that base is evaluated to C:\ so gulp.dest then creates the full path in the destination folder (Program Files (x86)\jenkins\...\.. ).

But it seems like glob issue. Check the base of the @stevelacy test with and without brackets in the folder name:
![screen shot 2016-08-23 at 14 40 40](https://cloud.githubusercontent.com/assets/1299001/17892074/afcf8a92-693f-11e6-8606-9b1c5ca3e9af.png)

**Comment by lukaskostial** [comment]
It seems that it is caused by isGlob and globParent implementation where isGlob evaluates such folder name as glob...

```js
module.exports = function isGlob(str) {
  return typeof str === 'string'
    && (/[*!?{}(|)[\]]/.test(str)
     || isExtglob(str));
};

module.exports = function globParent(str) {
	str += 'a'; // preserves full path in case of trailing path separator
	do {str = path.dirname(str)} while (isglob(str));
	return str;
};
```

**Comment by doowb** [comment]
I did a [PR to glob-stream](https://github.com/gulpjs/glob-stream/pull/67) that basically changes when the `basePath` is calculated to only check against the passed in glob.

This passes all the current tests and writes the `readme` to the correct location in the example repo.

**Comment by phated** [comment]
Thanks to the debugging by @lukaskostial and great work by @doowb this has been fixed in v5.3.3 of glob-stream
